src/charm.py

```diff
@@ -66,20 +66,20 @@ def __init__(self, *args: typing.Any) -> None:
         self.framework.observe(self.on.reset_instance_action, self._on_reset_instance_action)
         self.framework.observe(self.on.synapse_pebble_ready, self._on_pebble_ready)
 
-    def _change_config(self, event: ops.HookEvent) -> None:
-        """Change the configuration.
+    def change_config(self, event: ops.HookEvent) -> None:
+        """Change configuration.
 
         Args:
-            event: Event triggering the need of changing the configuration.
+            event: Event triggering after config needs to be changed.
         """
         container = self.unit.get_container(SYNAPSE_CONTAINER_NAME)
         if not container.can_connect():
             event.defer()
```

Comment:
I've header we should avoid deferring events. DK if there's any alternative here

Reply:
You're right. I found the [discussion](https://github.com/canonical/operator/issues/935#issue-1722066396) that @gregory-schiano shared on MM and it looks like event.defer() might generate a situation where you can have a queue with events waiting to be addressed and then, when the condition that the charm needs to continue is true, suddenly this could cause a lot of restarts, for example.
Unfortunately, I didn't find many examples of how to prevent that except for this one on [Alertmanager](https://github.com/canonical/alertmanager-k8s-operator/blob/main/src/charm.py#L380). It looks like a nice way of handling this so I'll do the same here.